src/Bridge/Doctrine/Orm/Filter/WhereFilter/Condition/AndCondition.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\WhereFilter\Condition;
+
+use Doctrine\ORM\Query\Expr;
+
+class AndCondition implements ConditionInterface
+{
+    public const TYPE = 'and';
+
+    /**
+     * @var ConditionInterface[]
+     */
+    protected $conditions;
+
+    public function __construct(ConditionInterface ...$conditions)
```

Comment:
We usually use an array instead of variadic parameters (it eases adding new parameter in next iterations).